File: 0003.无重复字符的最长子串/0003-无重复字符的最长子串.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
    
        string dic = "";
        int ans = 0;
        int i = 0;
        int j = 0;
        int l = s.size();
        while( i < l && j < l ){
            char a = s[i];
            char b = s[j];
            // cout<<dic.find(b)<<endl;
            if (dic.find(b) >= 0 && dic.find(b) < 99999999){
                i += 1;
                dic.erase(0, 1); 
            }
            else{
                j += 1;
                dic.push_back(b);
                // cout << "Dasdas" << endl;
            }
            ans = ans < j-i ? j-i : ans;
            // cout <<i<<" "<<j << endl;
        }
        return ans;
    }
};

File: 0003.无重复字符的最长子串/0003-无重复字符的最长子串.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        left, right = 0, 0
        dic = dict()
        res = 0
        while right < len(s):
            if s[right] in dic:
                left = max(left, dic[s[right]] + 1)
            dic[s[right]] = right
            res = max(res, right - left + 1)
            right += 1
        return res